loleaflet/src/control/Control.UIManager.js

```diff
@@ -166,9 +166,10 @@ L.Control.UIManager = L.Control.extend({
 			L.DomUtil.remove(L.DomUtil.get('spreadsheet-toolbar'));
 			L.DomUtil.remove(L.DomUtil.get('presentation-controls-wrapper'));
 
-			if ((window.mode.isTablet() || window.mode.isDesktop()) && this.map.isPermissionEdit()) {
-				var showRuler = this.getSavedStateOrDefault('ShowRuler');
-				L.control.ruler({position:'topleft', interactive:true, showruler: showRuler}).addTo(this.map);
+			if ((window.mode.isTablet() || window.mode.isDesktop())) {
+				var showRuler = window.docPermission === 'readonly' ? false : this.getSavedStateOrDefault('ShowRuler');
```

Comment:
I think maybe we can move this condition inside the `if` conditions. because if it is false we don't show ruler.